src/command_modules/azure-cli-network/azure/cli/command_modules/network/commands.py

```diff
@@ -331,16 +332,32 @@ def _make_singular(value):
     with self.command_group('network dns record-set') as g:
         g.custom_command('list', 'list_dns_record_set', client_factory=cf_dns_mgmt_record_sets, transform=transform_dns_record_set_output)
 
-    for record in ['a', 'aaaa', 'mx', 'ns', 'ptr', 'srv', 'txt', 'caa']:
+    api_version = str(get_api_version(self.cli_ctx, ResourceType.MGMT_NETWORK_DNS))
+    api_version = api_version.replace('-', '_')
+    dns_doc_string = 'azure.mgmt.dns.v' + api_version + '.operations#RecordSetsOperations.create_or_update'
```

Comment:
@tjprescott is this the standard approach to have different documentation for different profiles?

Reply:
I don't really think this is a case of have different documentation--it is just trying to get the docstrings from the SDK (which are probably the same) but it needs the profile name in the path due to the way multi-versioned SDKs are structured. I would actually argue for not using the `doc_string_source` argument and just adding the necessary help annotations.

Reply:
ditto

Reply:
hard coding the doc string breaks the code

Reply:
It means instead of trying to pull the docstring for this custom command to actually add an entries in `help.py` and `params.py` that populate the help.